### DISCUSSION TITLE
Display calculated property in Grid

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#6823] [General]
Hi all,

I've added the property EnabledDesc, read only, to my xyzRow.cs, that depends on value of property Enabled:

```
    [DisplayName("Abilitato"), NotNull]
    public short? Enabled
    {
        get => fields.Enabled[this];
        set => fields.Enabled[this] = value;
    }

    [DisplayName("Abilitato"), NotMapped]
    public string EnabledDesc
    {
        get
        {
            return Enabled == 1 ? "Si" : "No";
        }
    }
```

I added  _EnabledDesc_ to xyzColumns.cs too, in order to display it to the grid, but it is not resolved (see attached image)

I would like the property **Enabled** is used on edit form and property **EnableDesc** is used on drid

How can i do this; is there some tutorial?

Thank you.

![Schermata 2023-07-04 alle 15 00 57](https://github.com/serenity-is/Serenity/assets/14870650/0f67803a-404e-46c7-a51c-6445bfed6cc7)

**Comment by wezmag** [comment]
There are three options.
1. Use a formatter on the `Enabled` field.
2. Add Expression on the `EnabledDesc` field.
3. Populate `EnabledDesc` value in the ListHandler.

## Use a formatter on the `Enabled` field.
1. Add YesNoFormatter.ts in your project.
``` typescript
@Decorators.registerFormatter('YourProject.YourModule.YesNoFormatter')
export class YesNoFormatter implements Formatter {
    format(ctx: FormatterContext) {
        return ctx.value == 1 ? "Si" : "No";
    }
}
```
2. Apply `[YesNoFormatter]` attribute on the `Enabled` property in the column file.

If you choose to use a formatter, you don't need `EnabledDesc` anymore.


## Add Expression on the `EnabledDesc` field.
Change the `EnabledDesc` in your Row.cs file as below.

``` csharp
[DisplayName("Abilitato")]
[Expression("CASE WHEN T0.Enabled = 1 THEN 'Si' ELSE 'No' END")]
public string EnabledDesc
{
    get => fields.EnabledDesc[this];
    set => fields.EnabledDesc[this] = value;
}
```

## Populate `EnabledDesc` value in the ListHandler
1. Change the `EnabledDesc` in your Row.cs file as below.

``` csharp
[DisplayName("Abilitato")]
[NotMapped]
public string EnabledDesc
{
    get => fields.EnabledDesc[this];
    set => fields.EnabledDesc[this] = value;
}
```

2. Override the `OnReturn` in the corresponding ListHandler.cs

``` csharp
protected override void OnReturn()
{
   base.OnReturn();
   foreach (var entity in Response.Entities) {
      entity.EnabledDesc = entity.Enabled.HasValue && entity.Enabled.Value == 1 ? "Si" : "No";
   }
}
```

*Above code snippets are not tested! Please modify them accordingly.*

**Reply by wezmag** [reply]
In your case, it's recommended to use a formatter.

**Reply by elenaaralla** [reply]
Hi @wezmag,

the first option works perfectly!

What if I wold localize Yes and Not?

Thank you,
Elena.

**Reply by wezmag** [reply]
You can add keys in `Texts.cs` file.
for example:
``` csharp
public static partial class Texts
{
   public static partial class Site
   {
      public static class Common
      {
         public static LocalText Yes = "Yes";
         public static LocalText No = "No";
      }
   }
}
```
Then in your formatter, use the code below:
``` typescript
import { text } from "@serenity-is/corelib/q";
text("Site.Common.Yes"); //Yes
text("Site.Common.No"); //No
```
If `text` is not available in your version, I think you can use `localText` instead.

**Reply by elenaaralla** [reply]
Thank @wezmag,

it works fine!

What if i would localize the descriptions of an enum?

Thank you,
Elena.
